src/lib/services/tx/gql.ts

```diff
@@ -0,0 +1,102 @@
+import type { UseQueryResult } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
+import { useCallback } from "react";
+
+import { CELATONE_QUERY_KEYS, useCelatoneApp } from "lib/app-provider";
+import { getTxsByPoolIdPagination, getTxsCountByPoolId } from "lib/query";
+import { createQueryFnWithTimeout } from "lib/query-utils";
+import { usePoolTxExpression } from "lib/services/expression";
+import type {
+  BechAddr,
+  Message,
+  Option,
+  PoolTxFilter,
+  Transaction,
+} from "lib/types";
+import { ActionMsgType, MsgFurtherAction } from "lib/types";
+import { parseDate, parseTxHash, snakeToCamel } from "lib/utils";
+
+export const useTxsByPoolIdPagination = (
```

Comment:
@songwongtp Are we all in agreement about keeping hooks in `index.ts`?

Reply:
let's keep gql hooks here and export in `index.ts`, so that once we move all gql to api we can easily delete these hooks